hasUniqueChars.c

```diff
@@ -76,37 +76,34 @@ bool hasUniqueChars(char * inputStr) {
   // and checkBitsexcl_amp, for debugging purposes. 
   // It also illustrates how to use the seeBits function for debugging.
   // Printed values should initially be all zeros
-  // TODO: remove or comment out this code when satisfied of function correctness
   
-  char debug_str_A_z[128];
-  strcpy(debug_str_A_z, "checkBitsA_z before: \n");
-  seeBits(checkBitsA_z, debug_str_A_z);
+  // char debug_str_A_z[128];
+  // strcpy(debug_str_A_z, "checkBitsA_z before: \n");
+  // seeBits(checkBitsA_z, debug_str_A_z);
   
-  char debug_str_excl_amp[128];
-  strcpy(debug_str_excl_amp, "checkBitsexcl_amp before: \n");
-  seeBits(checkBitsexcl_amp, debug_str_excl_amp);
+  // char debug_str_excl_amp[128];
+  // strcpy(debug_str_excl_amp, "checkBitsexcl_amp before: \n");
+  // seeBits(checkBitsexcl_amp, debug_str_excl_amp);
   // -------------------------------------------------------------
-
-  // TODO: Declare additional variables you need here
-
   
-  for(i = 0; i < strlen(inputStr); i++) {
+  for(i = 0; i < strlen(inputStr); i++) { //loop through input string
     nextChar = inputStr[i];
-    // TODO: Add your code here to check nextChar, see if it is a duplicate, and update the checkBits variables
-
-    // -------------------------------------------------------------
-    // Below this are examples of debugging print statements you could use
-    // Move/use as makes sense for you!
-    // Modify to work on checkBitsexcl_amp
-    // TODO: Comment out or remove when your function works correctly
-    printf("nextchar int value: %d\n", nextChar);
-    char char_str[2] = "\0";
-    char_str[0] = nextChar;
-    strcpy(debug_str_A_z, "nextchar: ");
-    strcat(debug_str_A_z, char_str);
-    strcat(debug_str_A_z,", checkBitsA_z: \n");
-    seeBits(checkBitsA_z, debug_str_A_z);
-    // ------------------------------------------------------------- 
+  
+    //Check if char is a space, and whether to store it in checkBitsA_z or checkBitsexcl_amp.
+    if((int)nextChar < 65 && (int)nextChar != 32){
+      if(checkBitsexcl_amp & (long)1 << ((long)nextChar - 33)){  //Check if duplicate
+        return false;
+      }
+      //Update long to reflect that we've seen this character
+      checkBitsexcl_amp |= (long)1 << ((long)nextChar - 33);
```

Comment:
It's a good idea to avoid recomputing the same thing twice: assign a variable to it